src/esp/assets/ResourceManager.cpp

```diff
@@ -82,9 +83,10 @@ constexpr char ResourceManager::DEFAULT_LIGHTING_KEY[];
 constexpr char ResourceManager::DEFAULT_MATERIAL_KEY[];
 constexpr char ResourceManager::WHITE_MATERIAL_KEY[];
 constexpr char ResourceManager::PER_VERTEX_OBJECT_ID_MATERIAL_KEY[];
-ResourceManager::ResourceManager()
-    :
+ResourceManager::ResourceManager(Flags flags)
+    : flags_(flags)
```

Comment:
As we discussed during standup, the data these flags represent will eventually be managed on a per object/per stage basis via the MetadataMediator.